unit/src/symbol-table/symbol_table_util.cpp

```diff
@@ -0,0 +1,27 @@
+/*******************************************************************\
+
+ Module: Unit test utilities
+
+ Author: DiffBlue Limited. All rights reserved.
+
+\*******************************************************************/
+
+/// \file
+/// Utility functions relating to  the sybmol table.
+
+#include "symbol_table_util.h"
+
+/// Create a lvalue symbol of the specified type.
+/// \param name: The name of the symbol.
+/// \param type: The type of the symbol.
+/// \return The symbol that has been created.
+symbolt symbol_table_utilt::create_basic_symbol(
```

Comment:
Take a look at various `*_symbolt` classes defined in symbol.h - I'd see nothing wrong in adding one more, which really must be a useful one as otherwise you wouldn't have factored it out into a new function. (This would also address the namespace discussion in this PR.)